feat(markdown, json): add implementation for md and json format + update demo app

khushijoshy opened the pull request:
## Summary

Added support for markdown and json format to ngx-omniview.
Ensured **clean architecture** across the library and introduced a **professional JSON viewer** component.

This update ensures that all formats (text, HTML, Markdown, and JSON) render through a consistent registry-based system.
The JSON renderer now delivers a refined, minimal, and accessible viewing experience.

**Version:** 0.0.4
**Run demo:** `ng serve demo --open` → `http://localhost:4200/`
**Run tests:** `ng test ngx-omniview --watch=false --browsers=ChromeHeadless`

Currently supported formats: `text`, `html`, **`markdown`**, **`json`**

## Changes

### 1. Architecture & Core Fixes

- [x] Removed format-specific imports from main component
- [x] Replaced `if`-based checks with centralized **renderer registry**
- [x] Simplified template using `ngSwitch` for clean format handling
- [x] Standardized renderer outputs (string/object)
- [x] Moved `marked` and `ngx-markdown` to **peerDependencies**
- [x] Ensured Angular 15–20 compatibility

### 2. Markdown support

- [x] Sanitized rendering for text/html; markdown via `ngx-markdown`
- [x] Support for inline math and other formats inside markdown via `katex`

### 3. JSON Viewer

- [x] Fully redesigned JSON viewer (from scratch)
- [x] Implemented recursive rendering with expand/collapse toggles
- [x] Auto-collapse deep levels (≥2) with preview (`{ ... } (2 props)`)
- [x] Added item/property counts, hover effects, and transitions
- [x] Clean, neutral design — inherits theme, no forced colors
- [x] Aligned bracket structure, consistent indentation (16–20 px)
- [x] Accessible (ARIA labels, keyboard-friendly)
- [x] No external dependencies

### 4. Visual & UX

- [x] Unified typography and spacing across all renderers
- [x] Improved bracket alignment and toggle consistency
- [x] Introduced monospace font and smooth indentation guides
- [x] Responsive and adaptive to dark/light mode

### 5. Testing & Demo

- [x] Added tests for JSON, Markdown, and invalid input handling
- [x] All 9 tests passing (5 new)
- [x] Updated demo app with examples for text, HTML, Markdown, and JSON
- [x] Verified no console warnings or runtime errors

## Review Checklist

Please verify:

* [x] Demo loads at `http://localhost:4200/`
* [x] Text, HTML, Markdown, and JSON render correctly
* [x] Expand/collapse works smoothly in JSON viewer
* [x] Markdown and HTML sanitized correctly
* [x] All tests pass successfully
* [x] No architecture drift — registry pattern enforced

khushijoshy committed:
update: added html, markdown and json support

mratanusarkar reviewed (changes requested):
@khushijoshy some feedbacks:

- [x] library org pattern needs to be followed
- [x] demo app can be improved (as discussed)
- [x] json should have a viewer like [jsoneditoronline.org](https://jsoneditoronline.org/)

khushijoshy committed:
update: improvements

khushijoshy committed:
update: more improvements

khushijoshy committed:
update: json viewer updates

mratanusarkar reviewed (changes requested):
found some areas with the implementation that may be improved:

- [x] main component pollution
  format-specific modules are imported directly in the main component, example:
  ```ts
  import { MarkdownModule } from 'ngx-markdown';
  import { JsonViewerComponent } from './json-viewer/json-viewer.component';
  ```
- [x] registry pattern bypassed
  direct if-statements used instead of registry-based routing, example:
  ```ts
  if (this.format === 'html') return renderHtml(this.data);
  if (this.format === 'markdown') return renderMarkdown(this.data);
  if (this.format === 'json') { /* special logic */ }
  ```
- [x] template contains format logic
- [x] inconsistent return types
  renderers return string, but JSON viewer expects object — leads to architectural mismatch

mratanusarkar committed:
update: demo app sample data to reflect real usecase

mratanusarkar committed:
fix: package.json dependencies to peer dependency

mratanusarkar committed:
update: json renderer file

mratanusarkar committed:
update: clean the main component

- clean main component - Remove format-specific logic
- update template - Simple, clean structure
- keep JSON viewer - But integrate it properly
- keep markdown - But isolate the dependency
- use registry properly - All formats through one path

mratanusarkar committed:
add: create clean template structure with switch

mratanusarkar committed:
update: test cases for json and md

mratanusarkar committed:
add: new implementation of a simple, clean json viewer

mratanusarkar committed:
update: json-viewer to show collapsed from level 0

mratanusarkar committed:
update: sample json in demo app

mratanusarkar reviewed (approved):
Improved the PR and looks stable now!

mratanusarkar committed:
bump: version to 0.0.4

mratanusarkar committed:
update: package-lock

mratanusarkar committed:
update: npx npm-check-updates -u

mratanusarkar committed:
install: clean install and update

1) Clean root
```
rmdir /s /q node_modules
del /f /q package-lock.json
```

2) Clean the library too
```
rmdir /s /q projects\ngx-omniview\node_modules
del /f /q projects\ngx-omniview\package-lock.json
```

3) Fresh installs (root first, then library)
```
npm install
cd projects\ngx-omniview
npm install
cd ..\..
```

4) Apply Angular migrations if you bumped to v20
```
npx ng update @angular/core@20 @angular/cli@20
```

5) Optional: fix advisories
```
npm audit fix
```
